datastore/longhorn.go

```diff
@@ -4473,7 +4473,7 @@ func (s *DataStore) ListBackupVolumesWithVolumeNameRO(volumeName string) (map[st
 // GetBackupVolumeByBackupTargetAndVolumeRO returns a backup volume object using the given backup target and volume name in the cluster
 func (s *DataStore) GetBackupVolumeByBackupTargetAndVolumeRO(backupTargetName, volumeName string) (*longhorn.BackupVolume, error) {
 	if backupTargetName == "" || volumeName == "" {
-		return nil, fmt.Errorf("backup target name and volume name cannot be empty")
+		return nil, apierrors.NewNotFound(schema.GroupResource{Resource: "backupVolumes"}, "")
```

Comment:
The error seems misleading. It should be an error caused by the validation error. The original error looks clearer for me. Why do we change this?

Reply:
Reverted it.

>Why do we change this?

It's still an error but a `Not Found` error for the backup target or volume name is an empty string.